kasa/tests/fakeprotocol_smart.py

```diff
@@ -223,6 +223,20 @@ def _send_request(self, request_dict: dict):
             return retval
         elif method == "set_qs_info":
             return {"error_code": 0}
+        elif method == "set_dynamic_light_effect_rule_enable":
+            info["get_device_info"]["dynamic_light_effect_enable"] = params["enable"]
+            info["get_dynamic_light_effect_rules"]["enable"] = params["enable"]
+            if params["enable"]:
+                info["get_device_info"]["dynamic_light_effect_id"] = params["id"]
+                info["get_dynamic_light_effect_rules"]["current_rule_id"] = params[
+                    "enable"
+                ]
+            else:
+                if "dynamic_light_effect_id" in info["get_device_info"]:
+                    del info["get_device_info"]["dynamic_light_effect_id"]
+                if "current_rule_id" in info["get_dynamic_light_effect_rules"]:
+                    del info["get_dynamic_light_effect_rules"]["current_rule_id"]
```

Comment:
This feels overly complicated.. If it cannot be simplified anyway, it'd still make sense to put it into its own method.